addon/components/bs-modal.js

```diff
@@ -396,9 +368,7 @@ export default class Modal extends Component {
 
   @action
   doSubmit() {
-    // replace modalId by :scope selector if supported by all target browsers
-    let modalId = this.modalId;
-    let forms = this.modalElement.querySelectorAll(`#${modalId} .modal-body form`);
+    let forms = this.modalElement.querySelectorAll('.modal-body form');
```

Comment:
To be honest I'm not sure why modal ID (or `:scope`) was required here in the first place. But why do you thing it is not needed anymore?

Reply:
Well, the immediate reasons were that:

* `#some-id` selectors did not work inside Shadow DOM for me. Not sure why, I did not investigate.
* I removed the `modalId` entirely.

I made this specific change light hearted because I noticed that `modalId` pointed at `this.modalElement` anyway, so there was no reason to scope the selector by `this.modalElement` when searching inside `this.modalElement`. 🤷‍♂️

AFAIK, the only legit reason to reference the modal itself would be to use the immediate descendant combinator, e. g. `:scope > div`, because simply `> div` would be considered an invalid selector.

Also note that passing`":scope"` as `modalId` into the original code:

```
this.modalElement.querySelectorAll(`#${modalId} .modal-body form`);
```

...would end up as invalid selector `#:scope .modal-body form`.